homeassistant/components/devolo_home_control/sensor.py

```diff
@@ -95,3 +120,46 @@ def _sync(self, message=None):
         else:
             _LOGGER.debug("No valid message received: %s", message)
         self.schedule_update_ha_state()
+
+
+class DevoloConsumptionEntity(DevoloMultiLevelDeviceEntity):
+    """Representation of a consumption entity within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid, consumption):
+        """Initialize a devolo consumption sensor."""
+        self._device_instance = device_instance
+
+        self.value = getattr(
+            device_instance.consumption_property[element_uid], consumption
+        )
+        self.sensor_type = consumption
+        self.unit = getattr(
+            device_instance.consumption_property[element_uid], f"{consumption}_unit"
+        )
+        self.element_uid = element_uid
+
+        super().__init__(
+            homecontrol,
+            device_instance,
+            element_uid,
+            multi_level_sensor_property=self,
+            sync=self._sync,
+        )
```

Comment:
This pattern is repeated a bit in this integration.  I wonder if we could make this a bit more DRY (in another PR)

Reply:
We will take a look at it.